src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -507,15 +506,23 @@ private static void CheckParameterModifiers(BaseParameterSyntax parameter, Bindi
                 // error CS1100: Method '{0}' has a parameter modifier 'this' which is not on the first parameter
                 diagnostics.Add(ErrorCode.ERR_BadThisParam, thisKeyword.GetLocation(), owner.Name);
             }
-            else if (parameter.IsParams && owner.IsOperator())
+            else if (parameter.IsParams)
```

Comment:
>else if (parameter.IsParams) [](http://example.com/codeflow?start=12&length=28)

The refactoring doesn't look equivalent. It is disabling the following checks even if no errors reported. Consider moving this ```if``` to the end of the chain, or restructuring the code to perform the following checks unless an error is reported (perhaps ignoring the language version error).

Reply:
I believe the following checks are not applicable for a `params` parameter if no errors are reported here, although that may not be the case in the future, so I've moved this block to the end.